### PR TITLE
initial Sphinx manual conversion

### PR DESCRIPTION
**Pull request by cmills1095**
Conversion of the ASPECT manual to Sphinx online documentation. The workflow is that each chapter/section will go through two phases - in phase 1, the documentation is converted from tex to markdown with pandoc and then sorted into appropriate file structure corresponding with the pdf manual. However, the automatic conversion is not perfect and the files will need some 'cleaning up' (phase 2) to make sure references, equations, etc are formatted correctly.

Chapters 1 through 7 have phase 1 completed. Only chapter 1 (introduction) and sections 2.1 - 2.9 (Numerical methods) have undergone phase 2 and can be considered 'done' (pending approval).

### CODE REVIEW

**Comment by tjhei** [comment]
There are a bunch of files in ``_build``. Are they required to have, I would think not. Can you remove them?

File: doc/sphinx/conf.py [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+exclude_patterns = [
+    "_build",
+]
+
+# -- Options for HTML output -------------------------------------------------
+
+# The theme to use for HTML and HTML Help pages.  See the documentation for
+# a list of builtin themes.
+#
+html_logo = "_static/images/aspect_logo.png"
```
Comment (tjhei) [inline doc/sphinx/conf.py:60]: can we just reference the existing logo or does it need to be inside _static?

File: doc/sphinx/conf.py [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+html_sidebars = {
+    "**": ["search-field", "sidebar-nav-bs", "sidebar-ethical-ads", "sidebar-cig"]
+}
+bibtex_bibfiles = ["references.bib"]
+bibtex_default_style = "alpha"
+bibtex_reference_style = "author_year"
+numfig = True
+html_context = {
+    # "github_url": "https://github.com", # or your GitHub Enterprise interprise
+    "github_user": "cmills1095",
```
Comment (tjhei) [inline doc/sphinx/conf.py:86]: geodynamics?

File: doc/sphinx/make.bat [hunk 3]
```diff
@@ -0,0 +1,35 @@
+@ECHO OFF
```
Comment (tjhei) [inline doc/sphinx/make.bat:1]: this file won't be needed

File: doc/sphinx/conf.py [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+# import os
+# import sys
+# sys.path.insert(0, os.path.abspath('.'))
+
+
+# -- Project information -----------------------------------------------------
+
+project = 'ASPECT'
+copyright = '2022'
+author = 'Wolfgang Bangerth, Juliane Dannberg, Menno Fraters, Rene Gassmöller, Anne Glerum, Timo Heister, Bob Myhill, John Naliboff'
```
Comment (tjhei) [inline doc/sphinx/conf.py:22]: This is the list of code main developers, we will need the list of manual contributors here: https://github.com/geodynamics/aspect/blob/325d09ae4d3ea0634feb767828b31d305c785686/doc/manual/manual.tex#L188

File: doc/sphinx/conf.py [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+
+# The full version, including alpha/beta/rc tags
+release = '2.4.0-pre'
```
Comment (tjhei) [inline doc/sphinx/conf.py:25]: will need to parse ~/VERSION but this can wait until later.

**Comment by tjhei** [comment]
I think ``benchmarks/`` should not be inside ``cookbooks/`` but next to each other.

**Comment by gassmoeller** [comment]
@tjhei: Would you be ok with us merging this PR to have a rough structure in place to improve upon? Then I could also set it up as an automatic build (e.g. as actions workflow or as readthedocs build).

**Comment by gassmoeller** [comment]
@tjhei: Would you be ok with us merging this PR to have a rough structure in place to improve upon? Then I could also set it up as an automatic build (e.g. as actions workflow or as readthedocs build).

**Comment by tjhei** [comment]
Sure. Would it make sense to delete the _build/ and squash the commits to get rid of the files?

**Comment by gassmoeller** [comment]
We deleted _build already, but you have a good point, @cmills1095 please squash the commits in this branch into one, to get rid of the history (otherwise the intermediate commits that contain the large build files will still take up space in the repository). I can show you tomorrow how to do that if you need help.

**Comment by gassmoeller** [comment]
All tests that matter for this PR have finished, and the one still running is a duplicate of jenkins.tjhei.info. I will merge this for now so we can continue setting up the webhook to readthedocs (we can always disable that again if we decide to host ourselves instead).

**Pull request merged**